RecoHGCal/TICL/plugins/PFTICLProducer.cc

```diff
@@ -95,7 +119,30 @@ void PFTICLProducer::produce(edm::StreamID, edm::Event& evt, const edm::EventSet
       reco::TrackRef ref(ticl_cand.trackPtr().id(), int(ticl_cand.trackPtr().key()), &evt.productGetter());
       candidate.setTrackRef(ref);
     }
-    candidate.setTime(ticl_cand.time(), ticl_cand.timeError());
+
+    // HGCAL timing as default values
+    auto time = ticl_cand.time();
+    auto timeE = ticl_cand.timeError();
+
+    if (candidate.charge()) {  // Check MTD timing availability
+      const bool assocQuality = (*trackTimeQualH)[candidate.trackRef()] > timingQualityThreshold_;
+      if (assocQuality) {
+        const auto timeHGC = time;
+        const auto timeEHGC = timeE;
+        const auto timeMTD = (*trackTimeH)[candidate.trackRef()];
+        const auto timeEMTD = (*trackTimeErrH)[candidate.trackRef()];
+
+        if (useTimingAverage_ && (timeEMTD > 0 && timeEHGC > 0)) {
+          // Compute weighted average between HGCAL and MTD timing
+          timeE = sqrt(1 / (pow(timeEHGC, -2) + pow(timeEMTD, -2)));
+          time = (timeHGC / pow(timeEHGC, 2) + timeMTD / pow(timeEMTD, 2)) * pow(timeE, 2);
```

Comment:
Don't compute the `sqrt` of `timeE` right before having to square it again

Reply:
```suggestion
          const invTimeESqHGC = 1. / pow(timeEHGC, 2);
          const invTimeESqMTD = 1. / pow(timeEMTD, 2);
          timeE = invTimeESqHGC  * invTimeESqMTD / (invTimeESqHGC + invTimeESqMTD);
          time = (timeHGC * invTimeESqHGC + timeMTD * invTimeESqMTD)  * timeE;
          timeE = sqrt(timeE);
```

Reply:
Done:
https://github.com/cms-sw/cmssw/pull/32046/commits/65e52dd8eb097b00b185bf1f106e9c87ea99e053#diff-49ac77e92d14b3c79dfe08879ad1b7f4054eb34bf175e844010e4a77c0d242f1R132